include/dd/Package.hpp

```diff
@@ -52,6 +53,8 @@ namespace dd {
         static constexpr std::size_t UT_VEC_INITIAL_ALLOCATION_SIZE = 2048U;
         static constexpr std::size_t UT_MAT_NBUCKET                 = 32768U;
         static constexpr std::size_t UT_MAT_INITIAL_ALLOCATION_SIZE = 2048U;
+        static constexpr std::size_t UT_DM_NBUCKET                  = 32768U;
```

Comment:
How about changing the size of all tables to `1U` and increasing the size only when used. This would result in a small (but static) overhead at the beginning of simulation, but would make handling the table sizes quite a lot more convenient.

Reply:
I set the default size of density matrix caches to `1U` and added a special density matrix simulator ddPackage template with other parameters.